### PR TITLE
Telco 128 agw snap agw post install configuration

### PR DESCRIPTION
**Pull request by Gmerold**
This PR introduces Magma Access Gateway configurator. It allows user to configure installed AGW using the snap. 
Configuration steps:

- Copying Root CA PEM (copied from Orc8r installation) under Magma config dir
- Generating Control Proxy's configuration file
- Restarting Magma AGW services

Reference: [Magma docs](https://docs.magmacore.org/docs/next/lte/deploy_config_agw)

### CODE REVIEW

File: magma_access_gateway_configurator/agw_configurator.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        self._stop_service("magma@*")
+        self._restart_service("magma@magmad")
+
+    @property
+    def _control_proxy_config_dir_exists(self):
+        """Checks whether directory in which Proxy Config stores its configuration exists."""
+        return os.path.exists(self.MAGMA_CONTROL_PROXY_CONFIG_DIR)
+
+    def _create_control_proxy_config_dir(self):
+        """Creates directory in which Proxy Config stores its configuration."""
```
Comment (sanchezfdezjavier) [inline magma_access_gateway_configurator/agw_configurator.py:84]: Do we need this?

Comment (Gmerold) [reply magma_access_gateway_configurator/agw_configurator.py]: Sure we don't. 
Looks like some leftover after concept change ;)
Wonder why IDE didn't complain about it tho...

File: tests/unit/test_agw_configurator.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        mocked_path_exists.side_effect = [False, True]
+
+        with self.mocked_magma_control_proxy_config_dir:
+            self.agw_configurator.configure_control_proxy()
+
+        mocked_makedirs.assert_called_once_with(self.TEST_MAGMA_CONTROL_PROXY_CONFIG_DIR)
+
+    @patch("magma_access_gateway_configurator.agw_configurator.os.makedirs")
+    @patch("magma_access_gateway_configurator.agw_configurator.os.path.exists")
+    def test_given_existent_control_proxy_conf_dir_when_configure_control_proxy_then_control_proxy_conf_dir_is_not_created(  # noqa: E501
```
Comment (sanchezfdezjavier) [inline tests/unit/test_agw_configurator.py:100]: Why is the reason for this parameter order, why it shouldn't be `self, mocked_makedirs, mocked_path_exists`, as the first patch is `makedirs`. What am I missing here?

Comment (sanchezfdezjavier) [reply tests/unit/test_agw_configurator.py:100]: Nevermind

**Review by sanchezfdezjavier — APPROVED** [review]

File: tests/unit/test_agw_configurator.py (hunk 2 above)
Comment (sanchezfdezjavier) [reply tests/unit/test_agw_configurator.py:100]: My bad

File: tests/unit/test_agw_installer.py [hunk 3]
```diff
@@ -252,6 +252,7 @@ def test_given_magma_installation_process_when_start_magma_then_magma_services_a
 
     @patch("magma_access_gateway_installer.agw_installer.os.system")
     @patch("magma_access_gateway_installer.agw_installer.check_call", Mock())
+    @patch("magma_access_gateway_installer.agw_installer.check_output", MagicMock())
     @patch("magma_access_gateway_installer.agw_installer.open", mock_open())
     def test_given_magma_not_installed_when_install_then_system_goes_for_reboot_once_installation_is_done(  # noqa: E501
```
[the remark this answers is not in the document]
Comment (Gmerold) [reply tests/unit/test_agw_installer.py:258]: Agreed, fixed.

File: tests/unit/test_agw_configurator.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            new_callable=PropertyMock(return_value=self.TEST_MAGMA_CONTROL_PROXY_CONFIG_FILE_NAME),
+        )
+        self.agw_configurator = AGWConfigurator(self.TEST_DOMAIN, self.TEST_ROOT_CA_PEM_PATH)
+
+    @patch("magma_access_gateway_configurator.agw_configurator.os.makedirs")
+    @patch(
+        "magma_access_gateway_configurator.agw_configurator.shutil.copy",
+        Mock(),
+    )
+    def test_given_non_existent_root_ca_pem_destination_dir_when_copy_root_ca_pem_then_root_ca_pem_destination_dir_is_created(  # noqa: E501
```
[the remark this answers is not in the document]
Comment (Gmerold) [reply tests/unit/test_agw_configurator.py:49]: We definitely should. 
Fixed.

**Review by gruyaume — APPROVED** [review]

**Pull request merged**